.editorconfig

```
@@ -17,3 +17,8 @@ dotnet_sort_system_directives_first = true
 
 dotnet_diagnostic.IDE0073.severity = warning  # Require file header
 file_header_template = Copyright (c) Duende Software. All rights reserved.\nSee LICENSE in the project root for license information.
+
+dotnet_diagnostic.IDE0161.severity = error  # Use file-scoped namespace
+csharp_style_namespace_declarations = file_scoped:error
+
+dotnet_diagnostic.IDE0003.severity = error # Enforce that 'this.' should not be used when it's redundant
```

Title from https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0003-ide0009

```suggestion
dotnet_diagnostic.IDE0003.severity = error # Remove this or Me qualification
```